### ISSUE TITLE
Disallow undefined parameters

### ISSUE THREAD

**Issue by rlivsey** [ember-template-lint/ember-cli-template-lint#129]
This could be tricky because you need to know the context a template is operating in, and probably all helpers/components, but if I have the following in a template:

``` handlebars
{{foo}} and {{bar}}
```

And the following component:

``` javascript
export default Component.extend({
  foo: "here"
});
```

It should warn that `bar` is not defined.

**Comment by kellyselden** [comment]
This problem wouldn't be solvable here, as it only scans the template AST and not the js. You might have some luck with https://github.com/minichate/ember-cli-tree-shake

**Comment by Turbo87** [comment]
I'm flagging this as `wontfix` since it seems somewhat out-of-scope (for now) and hasn't been worked on in the past two years
